controllers/cloudflare_api.go

```diff
@@ -561,172 +426,121 @@ func (c *CloudflareAPI) GetDNSCNameId(fqdn string) (string, error) {
 	}
-
-	if len(dnsResponse.Result) == 0 {
-		c.Log.Info("no TXT records returned for fqdn", "fqdn", fqdn)
-		return "", DnsManagedRecordTxt{}, true, nil
-	}
-
-	var dnsTxtResponse DnsManagedRecordTxt
-	if err := json.Unmarshal([]byte(dnsResponse.Result[0].Content), &dnsTxtResponse); err != nil {
-		// TXT record exists, but not in JSON
-		c.Log.Error(err, "could not read TXT content in getting zoneId, check domain", "domain", c.Domain)
-		return dnsResponse.Result[0].Id, dnsTxtResponse, false, err
-	} else if dnsTxtResponse.TunnelId != c.ValidTunnelId {
-		// TXT record exists but not controlled by our tunnel
-		return dnsResponse.Result[0].Id, dnsTxtResponse, false, nil
-	}
-	return dnsResponse.Result[0].Id, dnsTxtResponse, true, nil
+	return "", DnsManagedRecordTxt{}, false, err
 }
 
 // InsertOrUpdateTXT upsert DNS TXT record for the given FQDN to point to the tunnel
 func (c *CloudflareAPI) InsertOrUpdateTXT(fqdn, txtId, dnsId string) error {
-	fqdn = TXT_PREFIX + fqdn
-	method := "POST"
-	subPath := ""
-	if txtId != "" {
-		c.Log.Info("Updating existing TXT record", "fqdn", fqdn, "dnsId", txtId)
-		method = "PUT"
-		subPath = "/" + txtId
-	} else {
-		c.Log.Info("Inserting DNS TXT record", "fqdn", fqdn)
-	}
-
 	content, err := json.Marshal(DnsManagedRecordTxt{
 		DnsId:      dnsId,
 		TunnelId:   c.ValidTunnelId,
 		TunnelName: c.ValidTunnelName,
 	})
 	if err != nil {
-		c.Log.Error(err, "could not marshal TXT record", "fqdn", fqdn)
-		return err
-	}
-
-	// Generate body for POST/PUT request
-	body, _ := json.Marshal(struct {
-		Type    string
-		Name    string
-		Content string
-		Ttl     int
-		Proxied bool
-	}{
-		Type:    "TXT",
-		Name:    fqdn,
-		Content: string(content),
-		Ttl:     1,     // Automatic TTL
-		Proxied: false, // TXT cannot be proxied
-	})
-	reqBody := bytes.NewBuffer(body)
-
-	req, _ := http.NewRequest(method, fmt.Sprintf("%szones/%s/dns_records%s", CLOUDFLARE_ENDPOINT, c.ValidZoneId, subPath), reqBody)
-	if err := c.addAuthHeader(req, false); err != nil {
+		c.Log.Error(err, "error marhsalling txt record json", "fqdn", fqdn)
 		return err
 	}
-	req.Header.Add("Content-Type", "application/json")
+	ctx := context.Background()
+	rc := cloudflare.ZoneIdentifier(c.ValidZoneId)
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		c.Log.Error(err, "error code in setting/updating DNS TXT record, check fqdn", "fqdn", fqdn)
-		return err
-	}
-
-	defer resp.Body.Close()
-	var dnsResponse CloudflareAPIResponse
-	if err := json.NewDecoder(resp.Body).Decode(&dnsResponse); err != nil {
-		c.Log.Error(err, "could not read body in setting DNS TXT record", "response", dnsResponse)
-		return err
-	} else if !dnsResponse.Success {
-		errs := ""
-		for _, errData := range dnsResponse.Errors {
-			errs += errData.Message
+	if txtId != "" {
+		c.Log.Info("Updating existing TXT record", "fqdn", fqdn, "dnsId", dnsId, "txtId", txtId)
+
+		updateParams := cloudflare.UpdateDNSRecordParams{
+			Type:    "TXT",
```

Comment:
This update needs an ID too.

```suggestion
		updateParams := cloudflare.UpdateDNSRecordParams{
			ID:      txtId,
			Type:    "TXT",
```